### PR TITLE
let "resolve" cope with foo.src packages

### PR DESCRIPTION
**Pull request by nphilipp**

### CODE REVIEW

**Comment by asamalik** [comment]
Tested, works for me.

This could be used to determine what needs to be present in a buldroot for a given set of source pacakes.

File: depchase [hunk 1]
```diff
@@ -356,14 +356,17 @@ def solve(solver, pkgnames, selfhost=False):
     jobs = []
     # Initial jobs, no conflicting packages
     for n in pkgnames:
-        sel = pool.select(n, solv.Selection.SELECTION_NAME | solv.Selection.SELECTION_DOTARCH)
+        sel = pool.select(n, solv.Selection.SELECTION_NAME
+                          | solv.Selection.SELECTION_DOTARCH
+                          | solv.Selection.SELECTION_WITH_SOURCE)
```
Comment (ignatenkobrain) [inline depchase]: no need for this, it works even without

File: depchase [hunk 2]
```diff
@@ -356,14 +356,17 @@ def solve(solver, pkgnames, selfhost=False):
...
         assert not sel.isempty(), "Could not find package for {}".format(n)
         jobs += sel.jobs(solv.Job.SOLVER_INSTALL)
     problems = solver.solve(jobs)
     assert not problems
 
     print_transaction(pool, solver.transaction())
-    candq = solver.transaction().newpackages()
+    candq = [s for s in solver.transaction().newpackages()
+             if s.arch not in ("src", "nosrc")]
```
Comment (ignatenkobrain) [inline depchase]: oneline please

File: depchase (hunk 1 above)
Comment (nphilipp) [inline depchase]: Ahh, I understood your comment before that it would be needed.

File: depchase (hunk 2 above)
Comment (nphilipp) [inline depchase]: Sure. What's your preferred line length limit, i.e. when to wrap lines?

Comment (ignatenkobrain) [inline depchase]: when it becomes unreadable (i.e. 100-120)

**Pull request merged**